Added first requirements

vladdord opened the pull request.

vladdord committed:
Added first requirements

bedrin commented on Requirements.md:
```diff
@@ -0,0 +1,72 @@
+ 3. Implementation of new database architecture with unique timestamps and three tables for predicted, historical and real-time data
+ 4. Redesign of UI for more convenient usage
+
+## Restrictions
```
Should go to non-functional requirements section

vladdord replied:
Removed and changed non-functional requirements

bedrin commented on Requirements.md:
```diff
@@ -0,0 +1,72 @@
+ 2. Prediction is not fully implemented
+ 3. System is able to store historical data
+
+## Plans for future
```
Remove it please - better create an issue ( https://github.com/RomanEgel/DeutscheBankProject/issues ) for each of them to track progress

All of them are already covered in requirements above (if not, it should go to requirements)

vladdord replied:
Done

bedrin commented on Requirements.md:
```diff
@@ -0,0 +1,72 @@
+    * [Intrinio](https://intrinio.com/)
+    * etc...
+
+## Current state
```
These items also should go to functional or non-functional requirements above

vladdord replied:
Removed

bedrin commented on Requirements.md:
```diff
@@ -0,0 +1,72 @@
+
+## Non-Functional requirements
+
+ 1. Project must be able to process data fast and accurate
```
We need numbers here. What is fast? sub-second? few seconds?

bedrin replied:
Other examples of NFR would be supported browsers (mobile?), concurrent users, availability, etc.

vladdord replied:
Numbers are very blurry and now close to random. We need to implement it first and then test performance

bedrin commented on Requirements.md:
```diff
@@ -0,0 +1,72 @@
+    * Data collector for real-time and historical data
+    * Predictor
+
+## Functional requirements
```
technical stuff (websokets, rest, spring, etc.) shouldn't be here - spring is not a requirement.
Suggest moving it to the bottom of the page to a new section called "Architecture" or something

vladdord replied:
Done

bedrin reviewed (changes requested):
Great spec. I would change a few bits - see my comments inline.

Probably can be renamed to README.md so it would show up on github when you open your project - as you wish

vladdord committed:
renamed Requirements.md -> README.md
Removed restrictions sections and updated non-functional requirements
Removed current state and plans for future sections
Re-wrote Functional requirements and added Architecture and techical details section

bedrin commented:
Great! Let's merge and I'll ask our FA's to review properly.
